File: app/src/main/java/com/scatl/uestcbbs/module/user/adapter/UserFriendAdapter.java
package com.scatl.uestcbbs.module.user.adapter;

import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.google.android.material.imageview.ShapeableImageView;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.UserFriendBean;
import com.scatl.uestcbbs.util.TimeUtil;

/**
 * author: sca_tl
 * description:
 * date: 2020/2/5 16:52
 */
public class UserFriendAdapter extends BaseQuickAdapter<UserFriendBean.ListBean, BaseViewHolder> {

    public UserFriendAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, UserFriendBean.ListBean item) {
        helper.setText(R.id.item_user_friend_name, item.name)
                .setText(R.id.item_user_friend_last_login,
                        TimeUtil.formatTime(item.lastLogin, R.string.last_login_time, mContext));
        Glide.with(mContext).load(item.icon).into((ShapeableImageView) helper.getView(R.id.item_user_friend_icon));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/user/adapter/UserFriendAdapter.kt
package com.scatl.uestcbbs.module.user.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.databinding.ItemUserFriendBinding
import com.scatl.uestcbbs.entity.UserFriendBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.TimeUtil
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/9 17:31
 */
class UserFriendAdapter: PreloadAdapter<UserFriendBean.ListBean, ItemUserFriendBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemUserFriendBinding {
        return ItemUserFriendBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemUserFriendBinding>, position: Int, item: UserFriendBean.ListBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.name.text = item.name
        holder.binding.lastLogin.text = TimeUtil.formatTime(item.lastLogin, R.string.last_login_time, context)
        holder.binding.avatar.load(item.icon)
    }
}